.vscode/settings.json

```diff
@@ -1,7 +1,8 @@
 {
     "[python]": {
         "editor.defaultFormatter": "ms-python.black-formatter",
-        "editor.formatOnSave": true
+        "editor.formatOnSave": true,
+        "editor.formatOnSaveMode": "modifications"
```

Comment:
This one is to help contributors to create ”clean” pull requests that are easier to review. `formatOnSaveMode": "modifications"` means that, when the file is saved, `black` only reformats the `git diff` (the changes to the file), rather than the whole file. A few times I found that saving a file made several formatting changes to areas of the file that I had not changed, and then I was manually undoing those changes in order to create pull requests that focus on the intended code changes.